internal/planner/average.go

```diff
@@ -102,7 +106,7 @@ func (n *averageNode) Next() (bool, error) {
 		return false, client.NewErrUnhandledType("sum", sumProp)
 	}
 
-	return true, nil
+	return mapper.RunFilter(n.currentValue, n.aggregateFilter)
```

Comment:
question: Why are you running the filter in the aggregate nodes, and not the select nodes they sit on top of (that also run the aggregate filters)?

Reply:
Or, is this more like a scan level filter?  And it is filtering the value produced by the aggregate node?  In which case would it not be better to run the filter in the select node that sits on top of the aggregates, allowing the aggregate nodes to remain focused on aggregating, and not adding an additional code-location in which filters may execute?

Or perhaps we could argue that the select node is doing too much, and we should in fact remove filtering from the Select node's remit, and instead introduce a filter node, removing filtering from select and aggregate nodes.

Reply:
I'm on board with moving it into a new `Filter` node but not within this same PR.

Reply:
What is the rush on this feature?  IMO refactors like this should be done before the feature, not after.  We need to think about the long term implications of how we build features before we build them.

Reply:
There's no rush. I'm trying to keep my PRs more focused, so I'll be doing the refactor in a different PR.